src/Sil/Component/Contact/Model/Contact.php

```diff
@@ -212,19 +212,71 @@ public function setDefaultAddress(AddressInterface $defaultAddress): void
     /**
      * Get the value of addresses.
      *
-     * @return Collection|AddressInterface[]
+     * @return array|AddressInterface[]
      */
-    public function getAddresses(): Collection
+    public function getAddresses(): array
     {
-        return $this->addresses;
+        return $this->addresses->toArray();
+    }
+
+    /**
+     * Add an Address to the collection.
+     *
+     * @param AddressInterface $address
+     */
+    public function addAddress(AddressInterface $address)
+    {
+        if ($this->addresses->contains($address)) {
+            throw new \InvalidArgumentException('This contact already owns this address');
+        }
+
+        $this->addresses->add($address);
+
+        if (!$this->defaultAddress) {
+            $this->setDefaultAddress($address);
+        }
+    }
+
+    /**
+     *  Remove an Address from the collection.
+     *
+     * @param AddressInterface $address
+     */
+    public function removeAddress(AddressInterface $address)
+    {
+        if (!$this->hasAddress($address)) {
+            throw new \InvalidArgumentException('Trying to remove an address that does not belong to this contact');
+        }
+
+        $this->addresses->removeElement($address);
+
+        if ($this->addresses->count() > 0) {
```

Comment:
You should check if `defaultAddress` is `null` before setting it with first address